packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useKeymap.ts

```diff
@@ -46,62 +84,49 @@ export default function useKeymap(CodeMirror: any) {
 			'Esc': 'singleSelection',
 		};
 
-		if (shim.isMac()) {
-			CodeMirror.keyMap.default = {
-				// MacOS
-				'Cmd-A': 'selectAll',
-				'Cmd-D': 'deleteLine',
-				'Cmd-Z': 'undo',
-				'Shift-Cmd-Z': 'redo',
-				'Cmd-Y': 'redo',
-				'Cmd-Home': 'goDocStart',
-				'Cmd-Up': 'goDocStart',
-				'Cmd-End': 'goDocEnd',
-				'Cmd-Down': 'goDocEnd',
-				'Cmd-Left': 'goLineLeft',
-				'Cmd-Right': 'goLineRight',
-				'Alt-Left': 'goGroupLeft',
-				'Alt-Right': 'goGroupRight',
-				'Alt-Backspace': 'delGroupBefore',
-				'Alt-Delete': 'delGroupAfter',
-				'Cmd-[': 'indentLess',
-				'Cmd-]': 'indentMore',
-				'Cmd-/': 'toggleComment',
-				'Cmd-Opt-S': 'sortSelectedLines',
-				'Opt-Up': 'swapLineUp',
-				'Opt-Down': 'swapLineDown',
-
-				'fallthrough': 'basic',
-			};
-		} else {
-			CodeMirror.keyMap.default = {
-				// Windows/linux
-				'Ctrl-A': 'selectAll',
-				'Ctrl-D': 'deleteLine',
-				'Ctrl-Z': 'undo',
-				'Shift-Ctrl-Z': 'redo',
-				'Ctrl-Y': 'redo',
-				'Ctrl-Home': 'goDocStart',
-				'Ctrl-End': 'goDocEnd',
-				'Ctrl-Up': 'goLineUp',
-				'Ctrl-Down': 'goLineDown',
-				'Ctrl-Left': 'goGroupLeft',
-				'Ctrl-Right': 'goGroupRight',
-				'Alt-Left': 'goLineStart',
-				'Alt-Right': 'goLineEnd',
-				'Ctrl-Backspace': 'delGroupBefore',
-				'Ctrl-Delete': 'delGroupAfter',
-				'Ctrl-[': 'indentLess',
-				'Ctrl-]': 'indentMore',
-				'Ctrl-/': 'toggleComment',
-				'Ctrl-Alt-S': 'sortSelectedLines',
-				'Alt-Up': 'swapLineUp',
-				'Alt-Down': 'swapLineDown',
-
-				'fallthrough': 'basic',
-			};
+		const defaultKeymap: CodeMirrorKey[] = [
+			// Windows/linux
+			{ label: _('Delete line'), command: 'editor.codemirror.deleteLine', default: 'Ctrl+D', macos: 'Cmd+D' },
+			{ label: _('Undo'), command: 'editor.codemirror.undo', default: 'Ctrl+Z', macos: 'Cmd+Z' },
+			{ label: _('Redo'), command: 'editor.codemirror.redo', default: 'Shift+Ctrl+Z', macos: 'Shift+Cmd+Z' },
+			{ label: _('Redo'), command: 'editor.codemirror.redo', default: 'Ctrl+Y', macos: 'Cmd+Y' },
```

Comment:
The "Redo" command appears twice, is that expected?

Reply:
Yes, Redo is commonly mapped to both Shift+Ctrl+Z and Ctrl+Y, this is how it worked previously and is also how it currently works in the WYSIWYG editor.
I still haven't made the updates I talked about earlier, but when I do I'll try to make sure this doesn't conflict. I'll just remove it if it does.